File: mobile/lib/models/identity_request.dart
class IdentityRequest {
  final String id;
  final String? makerId;
  final String? makerName;
  final String? makerEmail;
  final String fullName;
  final int age;
  final String gender;
  final String mobile;
  final String email;
  final String address;
  final String aadhaarFrontUrl;
  final String? aadhaarBackUrl;
  final String? passportUrl;
  final String status;
  final String remarks;
  final String? verifiedById;
  final String? verifiedByName;
  final DateTime? verifiedAt;
  final DateTime createdAt;

  IdentityRequest({
    required this.id,
    this.makerId,
    this.makerName,
    this.makerEmail,
    required this.fullName,
    required this.age,
    required this.gender,
    required this.mobile,
    required this.email,
    required this.address,
    required this.aadhaarFrontUrl,
    this.aadhaarBackUrl,
    this.passportUrl,
    required this.status,
    this.remarks = '',
    this.verifiedById,
    this.verifiedByName,
    this.verifiedAt,
    required this.createdAt,
  });

  factory IdentityRequest.fromJson(Map<String, dynamic> json) {
    // Handle populated makerId (can be string or object)
    String? makerId;
    String? makerName;
    String? makerEmail;
    if (json['makerId'] is Map) {
      makerId = json['makerId']['_id'];
      makerName = json['makerId']['name'];
      makerEmail = json['makerId']['email'];
    } else {
      makerId = json['makerId'];
    }

    // Handle populated verifiedBy
    String? verifiedById;
    String? verifiedByName;
    if (json['verifiedBy'] is Map) {
      verifiedById = json['verifiedBy']['_id'];
      verifiedByName = json['verifiedBy']['name'];
    } else {
      verifiedById = json['verifiedBy'];
    }

    return IdentityRequest(
      id: json['_id'] ?? '',
      makerId: makerId,
      makerName: makerName,
      makerEmail: makerEmail,
      fullName: json['fullName'] ?? '',
      age: json['age'] ?? 0,
      gender: json['gender'] ?? 'other',
      mobile: json['mobile'] ?? '',
      email: json['email'] ?? '',
      address: json['address'] ?? '',
      aadhaarFrontUrl: json['aadhaarFrontUrl'] ?? '',
      aadhaarBackUrl: json['aadhaarBackUrl'],
      passportUrl: json['passportUrl'],
      status: json['status'] ?? 'pending',
      remarks: json['remarks'] ?? '',
      verifiedById: verifiedById,
      verifiedByName: verifiedByName,
      verifiedAt: json['verifiedAt'] != null
          ? DateTime.tryParse(json['verifiedAt'])
          : null,
      createdAt: json['createdAt'] != null
          ? DateTime.parse(json['createdAt'])
          : DateTime.now(),
    );
  }

  bool get isPending => status == 'pending';
  bool get isVerified => status == 'verified';
  bool get isRejected => status == 'rejected';

  String get genderDisplay =>
      gender.isNotEmpty ? gender[0].toUpperCase() + gender.substring(1) : 'N/A';

  String get statusDisplay =>
      status[0].toUpperCase() + status.substring(1);
}

File: server/src/models/IdentityRequest.js
const mongoose = require('mongoose');

const identityRequestSchema = new mongoose.Schema(
  {
    makerId: {
      type: mongoose.Schema.Types.ObjectId,
      ref: 'User',
      required: true,
    },
    fullName: {
      type: String,
      required: [true, 'Full name is required'],
      trim: true,
      minlength: [2, 'Full name must be at least 2 characters'],
    },
    age: {
      type: Number,
      required: [true, 'Age is required'],
      min: [1, 'Age must be at least 1'],
      max: [150, 'Age cannot exceed 150'],
    },
    gender: {
      type: String,
      required: [true, 'Gender is required'],
      enum: ['male', 'female', 'other'],
    },
    mobile: {
      type: String,
      required: [true, 'Mobile number is required'],
      trim: true,
      unique: true,
      match: [/^[0-9]{10}$/, 'Please provide a valid 10-digit mobile number'],
    },
    email: {
      type: String,
      required: [true, 'Email is required'],
      trim: true,
      lowercase: true,
      unique: true,
      match: [/^\S+@\S+\.\S+$/, 'Please provide a valid email'],
    },
    address: {
      type: String,
      required: [true, 'Address is required'],
      trim: true,
    },
    aadhaarFrontUrl: {
      type: String,
      required: [true, 'Aadhaar front image is required'],
    },
    aadhaarBackUrl: {
      type: String,
      default: null,
    },
    passportUrl: {
      type: String,
      default: null,
    },
    status: {
      type: String,
      enum: ['pending', 'verified', 'rejected'],
      default: 'pending',
    },
    remarks: {
      type: String,
      default: '',
      trim: true,
    },
    verifiedBy: {
      type: mongoose.Schema.Types.ObjectId,
      ref: 'User',
      default: null,
    },
    verifiedAt: {
      type: Date,
      default: null,
    },
  },
  {
    timestamps: true,
  }
);

// Index for efficient queries
identityRequestSchema.index({ makerId: 1 });
identityRequestSchema.index({ status: 1 });
identityRequestSchema.index({ createdAt: -1 });
identityRequestSchema.index({ fullName: 'text' });

module.exports = mongoose.model('IdentityRequest', identityRequestSchema);
